ci(release): pnpm/action-setup errors on conflicting version + packageManager

## Symptom

Every push to `main` since the org transfer is failing the **Release** workflow at the `pnpm/action-setup@v4` step:

```
Error: Multiple versions of pnpm specified:
  - version 10 in the GitHub Action config with the key "version"
  - version pnpm@10.33.0 in the package.json with the key "packageManager"
Remove one of these versions to avoid version mismatch errors like ERR_PNPM_BAD_PM_VERSION
```

Failed run: https://github.com/danielsperoniteam/fhir-place/actions/runs/25603796590

## Cause

`pnpm/action-setup@v4` now hard-errors when both inputs are present. Two workflows still pass `with: version: 10` while `package.json` has `"packageManager": "pnpm@10.33.0"`:

- `.github/workflows/release.yml:26`
- `.github/workflows/pages.yml:56`

Other workflows (`ci.yml`, `integration.yml`, etc.) already use the bare action with no `with:` block and work fine.

## Fix

Drop the `with: version: 10` block from both workflow files. `packageManager` in `package.json` is the canonical source.

## Why high priority

Release is fully blocked. No release artifact has been published since the action started enforcing this. `pages.yml` deploys are also at risk — the failure is the same shape.

## Acceptance criteria

- [ ] `release.yml` succeeds on the next push to `main` after merge
- [ ] `pages.yml` continues to deploy `/` and `/staging/` correctly
- [ ] No other workflow regresses on the change

--- a/.github/workflows/pages.yml
+++ b/.github/workflows/pages.yml
@@ -54,8 +54,6 @@ jobs:
           path: staging-src
 
       - uses: pnpm/action-setup@v4
-        with:
-          version: 10
       - uses: actions/setup-node@v4
         with:
           node-version: 22

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -24,8 +24,6 @@ jobs:
           fetch-depth: 0
 
       - uses: pnpm/action-setup@v4
-        with:
-          version: 10
       - uses: actions/setup-node@v4
         with:
           node-version: 22

